shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/UTxO.hs

```diff
@@ -79,22 +83,37 @@ import Shelley.Spec.Ledger.TxData
     TxId (..),
     TxIn (..),
     TxOut (..),
+    UTxOOut (..),
     Wdrl (..),
     WitVKey (..),
     getRwdCred,
+    getAddress,
+    getAddressTx,
+    getValue,
+    getValueTx,
     pattern DeRegKey,
     pattern Delegate,
     pattern Delegation,
   )
 
--- | The unspent transaction outputs.
+-- needed for defining U
+-- | combine UTxOs
+-- TODO does it matter how we define this exactly? there is no right way
```

Comment:
Seems dodgy.

Reply:
What about the old instance? That relies on doing something with Maps, why won't that work here?

Reply:
This will be solved with upcoming Shelley changes to Relation class